nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/processors/standard/TransformXml.java

```diff
@@ -168,15 +168,18 @@ protected PropertyDescriptor getSupportedDynamicPropertyDescriptor(final String
 
     @OnScheduled
     public void onScheduled(final ProcessContext context) {
-        final Long cacheSize = context.getProperty(CACHE_SIZE).asLong();
-        final Long cacheDuration = context.getProperty(CACHE_DURATION).asTimePeriod(TimeUnit.SECONDS);
-
-        CacheBuilder cacheBuilder = CacheBuilder.newBuilder();
-        if (cacheSize > 0) {
-            cacheBuilder = cacheBuilder.maximumSize(cacheSize);
-            if (cacheDuration > 0) {
-                cacheBuilder = cacheBuilder.expireAfterAccess(cacheDuration, TimeUnit.SECONDS);
-            }
+        final ComponentLog logger = getLogger();
+        final Integer cacheSize = context.getProperty(CACHE_SIZE).asInteger();
+        final Long cacheTTL = context.getProperty(CACHE_TTL_AFTER_LAST_ACCESS).asTimePeriod(TimeUnit.SECONDS);
+
+        CacheBuilder cacheBuilder = CacheBuilder.newBuilder().maximumSize(cacheSize);
+
+        if (cacheSize <= 0) {
+            logger.warn("Stylesheet cache disabled because cache size is set to 0");
```

Comment:
The cache will still be created and on line 213 it will try to use it to look up an xslt if the cache size is 0. I'd suggest not even creating it in that case.